backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/client/impl/ElasticsearchClientFactoryImpl.java

```diff
@@ -129,6 +129,11 @@ public class ElasticsearchClientFactoryImpl implements ElasticsearchClientFactor
 			.asBeanReference( ElasticsearchHttpClientConfigurer.class )
 			.build();
 
+	private static final OptionalConfigurationProperty<Long> MAX_KEEP_ALIVE =
+			ConfigurationProperty.forKey( ElasticsearchBackendSettings.MAX_KEEP_ALIVE )
+					.asLongPositiveOrZero()
```

Comment:
Note that `0` is apparently interpreted as "keep alive forever"... That should probably be documented?

Reply:
hmm, from the docs:
> A value of 0 or less may be returned to indicate that there is no suitable suggestion.

that made me think that that would mean that it'll be up to a `ConnectionReuseStrategy` to do whatever it wants... 

Maybe it's better to just make our property strictly positive and not allow zero?